### PR TITLE
travis: Split all dist builders in two

### PR DESCRIPTION
**Pull request by alexcrichton**
Previously we would use one builder on Travis to produce two sets of host
compilers for two different targets. Unfortunately though we've recently
increased how much we're building for each target so this is starting to take
unnecessarily long (#40804). This commit splits the dist builders in two by
ensuring that we only dist one target on each builder, which should take a much
shorter amount of time. This should also unblock other work such as landing the
RLS (#40584).

### CODE REVIEW

File: src/ci/docker/run.sh [hunk 1]
```diff
@@ -45,20 +45,20 @@ else
     args="$args --env SCCACHE_DIR=/sccache --volume $HOME/.cache/sccache:/sccache"
 fi
 
-exec docker \
-  run \
```
Comment (aidanhs) [inline src/ci/docker/run.sh]: Indeed this is much faster, about 4min30sec total for the `ALLOW_PR` builder. But I'm not sure that disabling the `docker run` entirely is intentional?

Comment (alexcrichton) [inline src/ci/docker/run.sh]: Blast my attempt to improve our ci by 100000% percent has been foiled!

(definition not intentional to leave this change in)

File: src/ci/docker/dist-s390x-linux-netbsd/patches/glibc/2.12.2/001-Use-.machine-to-prevent-AS-from-complaining-about-z9.patch [hunk 2]
```diff
@@ -1,63 +0,0 @@
-From 2739047682590b1df473401b4febf424f857fccf Mon Sep 17 00:00:00 2001
```
Comment (cuviper) [inline src/ci/docker/dist-s390x-linux-netbsd/patches/glibc/2.12.2/001-Use-.machine-to-prevent-AS-from-complaining-about-z9.patch]: Did you mean to drop this s390x patch?

Comment (alexcrichton) [inline src/ci/docker/dist-s390x-linux-netbsd/patches/glibc/2.12.2/001-Use-.machine-to-prevent-AS-from-complaining-about-z9.patch]: Heh that'd explain why it failed on the bots, forgot we had a gitignore for `*.patch`

**Review by TimNN — APPROVED** [review]

**Comment by TimNN** [comment]
@bors r+

Approving since we need this to land #41045, which is currently blocking the queue.

**Comment by frewsxcv** [comment]
Failure on `IMAGE=dist-x86-linux`:

```
unable to prepare context: unable to evaluate symlinks in context path: lstat /home/travis/build/rust-lang/rust/src/ci/docker/dist-x86-linux: no such file or directory
```

https://travis-ci.org/rust-lang/rust/jobs/218515231

**Comment by frewsxcv** [comment]
Follow up from my last comment, this _should_ fix the issue:

https://github.com/rust-lang/rust/pull/41059/commits/b012adc354100ee95c1695ae7934e86683f1774e

**Pull request merged**